### PR TITLE
Tool for creating a github release

### PR DESCRIPTION
**Pull request by strekm**
**Description**

Changes proposed in this pull request:

- Tool for creating a github release

Part of #372

### CODE REVIEW

File: development/tools/cmd/githubrelease/main.go [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+var (
+	targetCommit           = flag.String("targetCommit", "", "Target commitish [Required]")
+	bucketName             = flag.String("bucketName", "kyma-prow-artifacts", "Google bucket name where artifacts are stored [Optional]")
+	kymaConfigCluster      = flag.String("kymaConfigCluster", "kyma-config-cluster.yaml", "Filename for cluster artifacts [Optional]")
+	kymaConfigLocal        = flag.String("kymaConfigLocal", "kyma-config-local.yaml", "Filename for local artifacts [Optional]")
+	kymaChangelog          = flag.String("kymaChangelog", "release-changelog.md", "Filename for release changelog [Optional]")
+	githubRepoOwner        = flag.String("githubRepoOwner", "", "Github repository owner [Required]")
+	githubRepoName         = flag.String("githubRepoName", "", "Github repository name [Required]")
+	githubAccessToken      = flag.String("githubAccessToken", "", "Github access token [Required]")
+	releaseVersionFilePath = flag.String("releaseVersionFilePath", "", "Full path to a file containing release version [Required]")
```
Comment (aszecowka) [inline development/tools/cmd/githubrelease/main.go]: to discuss: can we have parameter that contains directly release version, not path to file that contains it?

File: development/tools/pkg/file/file.go [hunk 2]
```diff
@@ -19,3 +20,21 @@ func FindAllRecursively(rootPath, extension string) ([]string, error) {
...
+		return "", err
+	}
+	defer file.Close()
+
+	data, err := ioutil.ReadAll(file)
+	if err != nil {
+		return "", err
+	}
+
+	content := string(data[:])
```
Comment (aszecowka) [inline development/tools/pkg/file/file.go]: ioutil.ReadAll can be used

File: development/tools/pkg/githubrelease/release.go [hunk 3]
```diff
@@ -0,0 +1,75 @@
+package githubrelease
+
+import (
+	"io/ioutil"
+	"os"
+)
+
+// Release .
+type Release struct {
+	Gap *GithubAPIWrapper
```
Comment (aszecowka) [inline development/tools/pkg/githubrelease/release.go]: Gap and Saw are not the best names IMO. Maybe we can rename it to Github and Storage?

File: development/tools/pkg/githubrelease/release.go [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+	Saw *StorageAPIWrapper
+}
+
+//CreateRelease .
+func (gr *Release) CreateRelease(releaseVersion string, targetCommit string, releaseChangelogName string, localArtifactName string, clusterArtifactName string, isPreRelease bool) error {
+	//kymaConfigCluster
+	clusterArtifactFile, err := gr.Saw.ReadBucketObject(clusterArtifactName)
+	if err != nil {
+		return err
```
Comment (aszecowka) [inline development/tools/pkg/githubrelease/release.go]: please wrap all errors to add info when it failed.

File: development/tools/pkg/githubrelease/release.go [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+	//clusterArtifactFile
+	_, _, err = gr.Gap.UploadArtifact(*release.ID, clusterArtifactName, clusterArtifactFile)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func readFile(filePath string) (string, error) {
```
Comment (aszecowka) [inline development/tools/pkg/githubrelease/release.go]: I already saw very similar function somewhere ;)

File: development/tools/pkg/githubrelease/storage.go [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+	if err != nil {
+		return nil, err
+	}
+
+	return saveArtifact(saw.TmpDir, fileName, data)
+}
+
+func saveArtifact(artifactsDir string, artifactFileName string, artifactData []byte) (*os.File, error) {
+	tmpArtifactFile, err := os.OpenFile(path.Join(artifactsDir, artifactFileName), os.O_RDWR|os.O_CREATE|os.O_EXCL, 0600)
+	if err != nil {
```
Comment (aszecowka) [inline development/tools/pkg/githubrelease/storage.go]: wraps errors

File: development/tools/pkg/githubrelease/storage.go [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+		return nil, err
+	}
+
+	_, err = tmpArtifactFile.Write(artifactData)
+	if err != nil {
+		return nil, err
```
Comment (aszecowka) [inline development/tools/pkg/githubrelease/storage.go]: I don't understand this method, why you close this file and then open it again.

File: development/tools/pkg/githubrelease/storage.go [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+	}
+
+	tmpArtifactFile, err = os.Open(tmpArtifactFile.Name())
+	if err != nil {
+		return nil, err
+	}
+
+	return tmpArtifactFile, nil
```
Comment (aszecowka) [inline development/tools/pkg/githubrelease/storage.go]: when this file will be closed?

File: development/tools/pkg/githubrelease/release.go [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+		return err
+	}
+
+	//changelog
+	releaseChangelogFile, err := gr.Saw.ReadBucketObject(releaseChangelogName)
+	if err != nil {
+		return err
+	}
+
+	releaseChangelogString, err := readFile(releaseChangelogFile.Name())
```
Comment (aszecowka) [inline development/tools/pkg/githubrelease/release.go]: why do you read this file one more time?

File: development/tools/cmd/githubrelease/README.md [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+```
+
+### Flags
+
+See the list of available flags:
+
+| Name                           | Required | Description                                                                                          |
+| :----------------------------- | :------: | :--------------------------------------------------------------------------------------------------- |
+| **--targetCommit**             |   Yes    | The string value with a commitish that the Github tag will refer to.
+| **--bucketName**               |    No    | The string value with a name of a Google bucket containing release artifacts. It defaults to `kyma-prow-artifacts`.
```
Comment (kazydek) [inline development/tools/cmd/githubrelease/README.md]:
```suggestion
| **--bucketName**               |    No    | The string value with the name of the Google bucket containing release artifacts. It defaults to `kyma-prow-artifacts`.
```

File: development/tools/cmd/githubrelease/README.md [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+| **--kymaConfigCluster**        |    No    | The string value with a name of a Kyma cluster configuration file. It defaults to `kyma-config-cluster.yaml`.
```
Comment (kazydek) [inline development/tools/cmd/githubrelease/README.md]:
```suggestion
| **--kymaConfigCluster**        |    No    | The string value with the name of the Kyma cluster configuration file. It defaults to `kyma-config-cluster.yaml`.
```

File: development/tools/cmd/githubrelease/README.md [hunk 12]
```diff
@@ -0,0 +1,44 @@
...
+| **--kymaConfigLocal**          |    No    | The string value with a name of a Kyma local configuration file. It defaults to `kyma-config-local.yaml`.
```
Comment (kazydek) [inline development/tools/cmd/githubrelease/README.md]:
```suggestion
| **--kymaConfigLocal**          |    No    | The string value with the name of the Kyma local configuration file. It defaults to `kyma-config-local.yaml`.
```

File: development/tools/cmd/githubrelease/README.md [hunk 13]
```diff
@@ -0,0 +1,44 @@
...
+| **--githubAccessToken**        |   Yes    | The string value with a name of the Github oath2 access token.
+| **--releaseVersionFilePath**   |   Yes    | The string value with the full path to the RELEASE_VERSION file.
+
+### Environment variables
+
+Available environment variables:
+
+| Name                                  | Required | Description                                                                                          |
+| :------------------------------------ | :------: | :--------------------------------------------------------------------------------------------------- |
+| **GOOGLE_APPLICATION_CREDENTIALS**    |    Yes   | The path to a service account file. The service account requires at least storage roles. |
```
Comment (kazydek) [inline development/tools/cmd/githubrelease/README.md]:
```suggestion
| **GOOGLE_APPLICATION_CREDENTIALS**    |    Yes   | The path to the service account file. The service account requires at least storage roles. |
```

File: development/tools/cmd/githubrelease/README.md [hunk 14]
```diff
@@ -0,0 +1,44 @@
...
+This command creates Github releases based on artifacts stored in a Google bucket. Each release requires the following set of artifacts:
+- `kyma-config-cluster.yaml`
+- `kyma-config-local.yaml`
+- `release-changelog.md`
+
+## Usage
+
+To run it, use:
+```bash
+env GOOGLE_APPLICATION_CREDENTIALS={path to a service account file} go run main.go \
```
Comment (kazydek) [inline development/tools/cmd/githubrelease/README.md]:
```suggestion
env GOOGLE_APPLICATION_CREDENTIALS={path to the service account file} go run main.go \ 
```

**Pull request merged**